src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20240103_143700.java

```diff
@@ -0,0 +1,415 @@
+        ExecutorService executorService = Executors.newFixedThreadPool(threadCount);
+
+        /*
+         * migrate the solution participations first, then the template participations, then the student participations
+         */
+        var solutionCount = solutionProgrammingExerciseParticipationRepository.count();
+        log.info("Found {} solution participations to migrate.", solutionCount);
+        for (int currentPageStart = 0; currentPageStart < solutionCount; currentPageStart += BATCH_SIZE) {
+            Pageable pageable = PageRequest.of(currentPageStart / BATCH_SIZE, BATCH_SIZE);
+            var solutionParticipationPage = solutionProgrammingExerciseParticipationRepository.findAll(pageable);
+            log.info("Will migrate {} solution participations in batch.", solutionParticipationPage.getNumberOfElements());
+            var solutionParticipationsPartitions = Lists.partition(solutionParticipationPage.toList(), threadCount);
+            for (var solutionParticipations : solutionParticipationsPartitions) {
+                executorService.submit(() -> migrateSolutions(solutionParticipations));
+            }
+        }
+
+        log.info("Submitted all solution participations to thread pool for migration.");
+        /*
+         * migrate the template participations
+         */
+        var templateCount = templateProgrammingExerciseParticipationRepository.count();
+        log.info("Found {} template participations to migrate", templateCount);
+        for (int currentPageStart = 0; currentPageStart < templateCount; currentPageStart += BATCH_SIZE) {
+            Pageable pageable = PageRequest.of(currentPageStart / BATCH_SIZE, BATCH_SIZE);
+            var templateParticipationPage = templateProgrammingExerciseParticipationRepository.findAll(pageable);
+            log.info("Will migrate {} template programming exercises in batch.", templateParticipationPage.getNumberOfElements());
+            var templateParticipationsPartitions = Lists.partition(templateParticipationPage.toList(), threadCount);
+            for (var templateParticipations : templateParticipationsPartitions) {
+                executorService.submit(() -> migrateTemplates(templateParticipations));
+            }
+        }
+
+        log.info("Submitted all template participations to thread pool for migration.");
+        /*
+         * migrate the student participations
+         */
+        log.info("Found {} student programming exercise participations with build plans to migrate.", studentCount);
+        for (int currentPageStart = 0; currentPageStart < studentCount; currentPageStart += BATCH_SIZE) {
+            Pageable pageable = PageRequest.of(currentPageStart / BATCH_SIZE, BATCH_SIZE);
+            Page<ProgrammingExerciseStudentParticipation> studentParticipationPage = ciMigrationService.orElseThrow()
+                    .getPageableStudentParticipations(programmingExerciseStudentParticipationRepository, pageable);
```

Comment:
Can we sort the student participations and start with the last ones (i.e. highest id). Old repos are typically not really interesting, because they are not used any more and we should start with the more important ones